### DISCUSSION TITLE
SolcInstallationError

### DISCUSSION THREAD

**Discussion by Jimmy321342** [smartcontractkit/full-blockchain-solidity-course-py#362] [Q&A] [answered]
![h9](https://user-images.githubusercontent.com/92639455/141734465-203d620f-1c53-411f-928b-61a107d97382.PNG)

updated my code with suggested changes on https://github.com/smartcontractkit/full-blockchain-solidity-course-py/blob/main/chronological-issues-from-video.md

still throwing an error, please help.

**Comment by cromewar** [comment, marked as answer]
Hello @Jimmy321342 I saw this problem[here](https://ethereum.stackexchange.com/questions/102866/how-to-compile-smart-contracts-using-brownie-on-raspberry-pi-4b?noredirect=1&lq=1) but it was for raspberry-py as you are on windows I  have two recommendations for you:

1. reinstall `pip install py-solc-x` in a python virutal environment, including eth-brownie, and try again (use [this](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/) as a guide for virtualenvs).
2. Use WSL (windows subsystem for linux) you may refer to [this](https://medium.com/@cromewar/how-to-setup-windows-10-11-for-smart-contract-development-and-brownie-e7d8d13555b3) guide.

**Reply by Jimmy321342** [reply]
reinstall pip install py-solc-x in a python virutal environment, including eth-brownie, and try again (use this as a guide for virtualenvs).  tried this,  throws following error

pip failed to build package:
    cytoolz

Some possibly relevant errors from pip install:
    build\lib.win-amd64-3.10\cytoolz\functoolz.cp310-win_amd64.pyd : fatal error LNK1120: 1 unresolved externals
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\link.exe' failed with exit code 1120

Error installing eth-brownie.

**Reply by Jimmy321342** [reply]
tried with both pip and pipx, it is still stuck

**Reply by cromewar** [reply]
OK we are on the correct way, the Cytoolz problem is common and some windows user are having it, you have two possible paths here:

1. install visual studio community and the python development tools on it.
2. install WSL and work with a Linux terminal instead.
